### DISCUSSION TITLE
More built-in commonly used Features for lexical, such as code blocks or tables

### DISCUSSION THREAD

**Discussion by otomir23** [payloadcms/payload#4112] [Feature Requests & Ideas]
`@payloadcms/lexical` is currently missing some commonly used Features (such as code blocks, tables, etc) so they have to be implemented from scratch if you project needs them. It would be nice if a standard implementation for them was built-in to `@payloadcms/lexical`.

**Comment by ssyberg** [comment]
Also interested in what the outlook/plan on this is! Tables is a big one.

**Comment by ChrisGV04** [comment]
I would also be very interested in seeing tables not only in Lexical, but as a component usable across Payload

**Comment by AlessioGr** [comment]
The current focus of the lexical editor is making sure it's 100% stable. As of now, before 1.0, no big new features like tables are planned.

Post-1.0, we're going to add new features - including tables! Until then, you could attempt making your own Table feature by using something like tanstack table and wrapping it in a lexical `DecoratorBlockNode`.

**Comment by hades200082** [comment]
I'm interested in code blocks, specifically ones where I can select the language so my front-end can use something like prism to render it with syntax highlights

**Comment by GermanJablo** [comment]
Since this discussion was opened, tables, codeblocks, and arbitrary blocks have been added. Please, if there is still something specific you would like to see in the Lexical editor, please open a specific feature request. Thanks!

**Reply by hades200082** [reply]
The only other one I can think of that would be useful is `iframe` for those times where I need to embed an external thing like Mailchimp/Pardot forms, etc.

**Reply by GermanJablo** [reply]
You can do that easily with a block! Inside you can define custom components

https://payloadcms.com/docs/rich-text/custom-features#example-code-field-block-with-language-picker

**Reply by hades200082** [reply]
You can do most of these things with custom blocks. The point of this was to ask for commonly used stuff to be added as out-of-the-box features right?
